- ✍🏼 `transformStateWithClones` function to handle ➕add, ❌remove, and…

de-vasta opened the pull request:
… 🧹clear (based on `actions`) props of the `state` object without reassigning function's params and returning the array of modified object of every action.

de-vasta committed:
✍🏼 `transformStateWithClones` function to handle ➕add, ❌remove, and 🧹clear (based on `actions`) props of the `state` object without reassigning function's params and returning the array of modified object of every action.
➕ simple `switch` `default:`

DemonWayne commented on src/transformStateWithClones.js:
```diff
@@ -7,7 +7,38 @@
+        }
+        break;
+      default:
+        throw new Error(`Unknown action type: ${action.type}`);
```
It’s better to wrap each `case` in `{}` in a `switch` statement

de-vasta replied:
Ok! The requested little adjustments: a4cd18a

de-vasta committed:
➕ Add curly braces to switch cases

DemonWayne reviewed (approved):
Good work!

ArturErmolenko commented on src/transformStateWithClones.js:
```diff
@@ -7,7 +7,43 @@
+  const REMOVE_ACTION = 'removeProperties';
+  const CLEAR_ACTION = 'clear';
+
+  // Array to hold clones of `state` after each action
```
```suggestion
```
remove comments

ArturErmolenko commented on src/transformStateWithClones.js:
```diff
@@ -7,7 +7,43 @@
+
+  // Array to hold clones of `state` after each action
+  const stateClones = [];
+  const prevClone = { ...state };
```
```suggestion
  const prevClone = { ...state };
```
why prev? It's looks like copy ;)

stateCopy sounds better for me

ArturErmolenko reviewed (approved):
Nice job 🔥

de-vasta committed:
🔧 Refactor:
- changed variable name from obscure `prevClone` to more suitable `stateCopy`;
- removed redundant comments
